src/hooks/cart/useGetCartCount.ts

```diff
@@ -0,0 +1,38 @@
+import {useEffect, useState} from 'react';
+import {useSelector} from 'react-redux';
+import {getWishlistState} from '../../redux/wishlist/selectors';
+import CartServices from '../../services/CartServices';
+
+const useGetCartCount = () => {
+  const [count, setCount] = useState(0);
+  const dataCart = useSelector(getWishlistState);
+
+  useEffect(() => {
+    if (dataCart.length === 0) {
+      CartServices.getCart()
+        .then(res => {
+          if (res.data.cartItems.length > 1)
+            setCount(
+              res.data.cartItems
+                .map(item => item.quantity)
+                .reduce((prv: number, cur: number) => {
+                  return prv + cur;
+                }),
+            );
+          else setCount(0);
+        })
+        .catch(err => console.log(err));
+    } else
+      setCount(
+        dataCart
+          .map(item => item.quantity)
+          .reduce((prv: number, cur: number) => {
+            return prv + cur;
+          }),
+      );
```

Comment:
không cần tốn thêm 1 lần map 
```suggestion
      setCount(
        dataCart
          .reduce((prv: number, item) => {
            return prv + item. quantity;
          }),
      );
```